File: Angular/src/app/Models/Classes/ItemTransaction.ts
import { Items } from "./Items";
import { Person } from "./Person";

export class ItemTransaction {
    id!: number;
    items?: Items;
    itemsId!: number;
    transactionDate?: Date;
    handOverPerson!: Person;
    handOverPersonId?: number;
    reciverPerson!: Person;
    reciverPersonId?: number;
    quantity!: number;
    paid!: number;
}

File: Nozom.Data/Entities/Storage/ItemTransaction.cs
using Nozom.Data.EntityInterfaces;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Nozom.Data.Entities.Storage
{
    public class ItemTransaction: IDbModel<int>
    {
        public int Id { get; set; }

        [ForeignKey("ItemsId")]
        public Items Items { get; set; }

        public int ItemsId { get; set; }

        public DateTime TransactionDate { get; set; }

        [ForeignKey("HandOverPersonId")]
        public Person HandOverPerson { get; set; }

        public int HandOverPersonId { get; set; }

        [ForeignKey("ReciverPersonId")]
        public Person ReciverPerson { get; set; }

        public int ReciverPersonId { get; set; }

        public int Quantity { get; set; }

        public decimal Paid { get; set; }
    }
}
